br/pkg/lightning/mydump/csv_parser.go

```diff
@@ -341,7 +342,7 @@ func (parser *CSVParser) readUntil(chars *byteSet) ([]byte, byte, error) {
 			if err == nil {
 				err = io.EOF
 			}
-			parser.pos += int64(len(buf))
+			// parser.pos += int64(len(buf))
```

Comment:
This changed the semantics of `pos`. `pos` means the current file offset. Since we have read to the end, the pos should be set to the end offset.

Reply:
just a test. I didn't change it here.

Reply:
This demand is from import on cloud. We are trying to use the offset reported in the syntax error to show the content in users' files. However, in an unterminated quote error, the offset always exceeds the range of the file. So I am about to return the position to the previous one.

If we meet an error, setting back the `pos` seems safe because we will never use it again.

Reply:
> However, in an unterminated quote error, the offset always exceeds the range of the file.

Why does offset exceed the range of the file?  The end offset is also a valid offset.

It's hard to tell which pos is the starting point of syntax error. Suppose we have a line of data: `a,b,c,"xxxxx`
It's obvious that it has an unterminated quote error. But what is the right pos of syntax error? Is it the start pos of `"xxxx` or the end pos of this line? In my opinion, the end pos is not a wrong answer, because everything is still valid until the end.

I'm not sure how you use the syntax error pos. If you just want to print content near the error, you can print the content around the pos. e.g When pos is 1000, print content[800:1200].

Reply:
We tend to present the starting point of the unterminated quote (i.e., the first quote). Because users should fix the data files and import them again, acknowledging the starting point would be better than telling them the end offset of the whole file, which is a bit meaningless.

Reply:
> Why does offset exceed the range of the file? The end offset is also a valid offset.

It is the position of EOF (e.g., the size of the file is 35 bytes, then the end of the offset is 35).